File: two-sum/two-sum-mustafasegf.go
package twosum

func twoSum(nums []int, target int) []int {
	nMap := map[int]int{}
	for i, n := range nums {
		if ii, ok := nMap[target-n]; ok {
			return []int{i, ii}
		}
		nMap[n] = i
	}

	return nil
}

File: two-sum/two-sum-mustafasegf.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();

        for (i, num) in nums.iter().enumerate() {
            match map.get(&(target - num)) {
                None => map.insert(num, i),
                Some(idx) => return vec![*idx as i32, i as i32],
            };
        }
        return vec![0, 0];
    }
}
